modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/internal/upgrade/v10_5_0/test/ObjectEntryDefaultLanguageIdUpgradeProcessTest.java

```diff
@@ -0,0 +1,217 @@
+import com.liferay.portal.test.rule.Inject;
+import com.liferay.portal.test.rule.LiferayIntegrationTestRule;
+import com.liferay.portal.test.rule.PermissionCheckerMethodTestRule;
+import com.liferay.portal.upgrade.registry.UpgradeStepRegistrator;
+import com.liferay.portal.upgrade.test.util.UpgradeTestUtil;
+import com.liferay.portal.vulcan.util.LocalizedMapUtil;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Set;
+
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+/**
+ * @author Pedro Tavares
+ */
+@RunWith(Arquillian.class)
+public class ObjectEntryDefaultLanguageIdUpgradeProcessTest {
+
+	@ClassRule
+	@Rule
+	public static final AggregateTestRule aggregateTestRule =
+		new AggregateTestRule(
+			new LiferayIntegrationTestRule(),
+			PermissionCheckerMethodTestRule.INSTANCE);
+
+	@Before
+	public void setUp() throws Exception {
+		_company = CompanyTestUtil.addCompany();
+		_group = GroupTestUtil.addGroup();
+		_originalName = PrincipalThreadLocal.getName();
+		_originalPermissionChecker =
+			PermissionThreadLocal.getPermissionChecker();
+	}
+
+	@After
+	public void tearDown() {
+		PermissionThreadLocal.setPermissionChecker(_originalPermissionChecker);
+		PrincipalThreadLocal.setName(_originalName);
+	}
+
+	@Test
+	public void testUpgrade() throws Exception {
+		User companyAdminUser = UserTestUtil.addCompanyAdminUser(_company);
+
+		PermissionThreadLocal.setPermissionChecker(
+			PermissionCheckerFactoryUtil.create(companyAdminUser));
+		PrincipalThreadLocal.setName(companyAdminUser.getUserId());
+
+		CompanyTestUtil.resetCompanyLocales(
+			_company.getCompanyId(),
+			Set.of(LocaleUtil.BRAZIL, LocaleUtil.SPAIN, LocaleUtil.US),
+			LocaleUtil.BRAZIL);
+
+		ObjectDefinition companyObjectDefinition =
+			ObjectDefinitionTestUtil.publishObjectDefinition(
+				ObjectDefinitionTestUtil.getRandomName(),
+				Collections.singletonList(
+					new TextObjectFieldBuilder(
+					).labelMap(
+						LocalizedMapUtil.getLocalizedMap(
+							RandomTestUtil.randomString())
+					).name(
+						"textObjectField"
+					).build()),
+				ObjectDefinitionConstants.SCOPE_COMPANY,
+				companyAdminUser.getUserId());
+
+		ObjectEntry companyObjectEntry =
+			_objectEntryLocalService.addObjectEntry(
+				companyAdminUser.getUserId(), 0, null,
+				companyObjectDefinition.getObjectDefinitionId(),
+				Collections.singletonMap(
+					"textObjectField", RandomTestUtil.randomString()),
+				ServiceContextTestUtil.getServiceContext());
+
+		companyObjectEntry.setDefaultLanguageId(null);
+
+		companyObjectEntry = _objectEntryLocalService.updateObjectEntry(
+			companyObjectEntry);
+
+		Assert.assertEquals(
+			_company.getCompanyId(), companyObjectEntry.getCompanyId());
+		Assert.assertTrue(
+			Validator.isNull(companyObjectEntry.getDefaultLanguageId()));
```

Comment:
Use `Assert.assertNull()`, please.

Reply:
`Assert.assertNull()` won't work here, because the defaultLanguageId comes as an empty string. So I did this way so that both null and empty string will be okay.